app/components/story/StoryCard.js

```diff
@@ -4,18 +4,48 @@ import {
   Text,
   StyleSheet,
   Dimensions,
+  Animated,
 } from 'react-native';
 
 class StoryCard extends Component {
+  constructor(props) {
+    super(props); 
+    this.state = {
+      height: new Animated.Value() 
+    };
+  }
+
+  componentWillMount() {
+    if (this.props.isSplit) {
+      this.state.height.setValue(Dimensions.get('window').height/2);
+    } else {
+      this.state.height.setValue(0);
+    }
+  }
+
+  componentWillReceiveProps(nextProps) {
+    if (nextProps.isSplit != this.props.isSplit) {
+      Animated.timing(
+        this.state.height,
+        { 
+          toValue: this.props.isSplit ? 0 : Dimensions.get('window').height/2,
```

Comment:
this height value will be constant and should be memoized instead of being recalculated every time